#!/usr/bin/env python
PACKAGE = "fmExecutors"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

rabbit_type_enum = gen.enum([ gen.const("Auto", int_t, 0, "auto mode"),
                  gen.const("Fixed", int_t, 1, "a fixed rabbit x meters in front"),
                  gen.const("Float", int_t, 2, "a floating rabbit scaled to the line length")],
                  "An enum controlling the planning mode of the rabbitPlanner")
                  

gen.add("Rabbit_type",       int_t, 0, "The type of rabbit to place", 0, 0, 2, edit_method=rabbit_type_enum)
gen.add("Distance_scale", double_t, 0, "For automode, the amount to factor in the distance error to the line", 0.1, 0, 10)
gen.add("Angle_scale",    double_t, 0, "For automode, the amount to factor in the angle error to the line",1,0,10)
gen.add("Delta_rabbit",   double_t, 0, "for fixed and float mode, the amount to scale the rabbit",4,0.1,30)
gen.add("Delta_waypoint", double_t, 0, "distance to waypoint B before switching to next",0.1,0,2)


exit(gen.generate(PACKAGE, "dynamic_tutorials", "rabbitPlanner"))